app/containers/EmojiPicker/EmojiCategory.tsx

```diff
@@ -26,48 +28,44 @@ const renderEmoji = (emoji: IEmoji, size: number, baseUrl: string) => {
 	);
 };
 
-class EmojiCategory extends React.Component<IEmojiCategory> {
-	renderItem(emoji: IEmoji) {
-		const { baseUrl, onEmojiSelected } = this.props;
-		return (
+const EmojiCategory = React.memo(
+	({ baseUrl, onEmojiSelected, emojis, width, tabsCount, isBottomSheet, ...props }: IEmojiCategory) => {
+		const emojiSize = width ? Math.min(width / tabsCount, MAX_EMOJI_SIZE) : MAX_EMOJI_SIZE;
+		const numColumns = Math.trunc(width ? width / emojiSize : tabsCount);
+
+		const FlatList = isBottomSheet ? BottomSheetFlatList : GHFlatList;
+
+		const renderItem = (emoji: IEmoji) => (
 			<TouchableOpacity
 				activeOpacity={0.7}
 				// @ts-ignore
 				key={emoji && emoji.isCustom ? emoji.content : emoji}
 				onPress={() => onEmojiSelected(emoji)}
 				testID={`reaction-picker-${emoji && emoji.isCustom ? emoji.content : emoji}`}>
-				{renderEmoji(emoji, EMOJI_SIZE, baseUrl)}
+				{renderEmoji(emoji, emojiSize, baseUrl)}
```

Comment:
We should turn this into a component.
The way you're using (`renderEmoji()`), is a function and not a component, so React is not aware of it, it doesn't know what to expect as the result and so it needs to re-evaluate it every time.
As a component (`<EmojiSomething />`), React will be able to reconcile it, making use of the props appropriately.